acq4/util/imaging/frame.py

```diff
@@ -70,26 +107,45 @@ def mapFromFrameToGlobal(self, obj):
         """Map *obj* from the frame's data coordinates to global coordinates.
         """
         return self.globalTransform().map(obj)
-    
-    def saveImage(self, dh, filename, backgroundInfo: Optional[Callable] = None, contrastInfo=None):
+
+    @property
+    def time(self):
+        return self._info['time']
+
+    @property
+    def depth(self):
+        return self.mapFromFrameToGlobal(pg.Vector(0, 0, 0)).z()
+
+    def saveImage(self, dh, filename=None, appendTo=None, appendAxis=None, autoIncrement=True):
         """Save this frame data to *filename* inside DirHandle *dh*.
 
         The file name must end with ".ma" (for MetaArray) or any supported image file extension.
-        The optional *backgroundInfo* and *contrastInfo* arguments can be used to save additional
-        data needed to display the frame later. See `BackgroundSubtractCtrl.defferedSave` and
-        `ContrastCtrl.saveState` for details.
+
+        If *appendTo* is not None, the file will be appended to *appendTo* along the *appendAxis*, which
+        value we will supply from this object (e.g. "Depth" goes to `self.depth`).
         """
         data = self.getImage()
         info = self.info()
-        if backgroundInfo is not None:
-            info['backgroundInfo'] = backgroundInfo(dh)
-        if contrastInfo is not None:
-            info['contrastInfo'] = contrastInfo
+        if callable(info.get('backgroundInfo')):
+            info['backgroundInfo'] = info['backgroundInfo'](dh)
+        if filename and not filename.endswith('.ma'):
+            return dh.writeFile(data, filename, info, fileType="ImageFile", autoIncrement=autoIncrement)
+
+        if appendAxis:
+            array_info = [
+                {'name': appendAxis, 'values': [getattr(self, appendAxis.lower())]},
```

Comment:
this is ugly

Reply:
we could just require the arg to be the array_info entry, rather than just the string name.

Reply:
that would look like e.g.: `frame.saveImage(..., appendAxis={'name': 'Depth', 'values': [frame.depth]})`